### PR TITLE
Correct time units for `BlackrockSortingExtractor`

### PR DESCRIPTION
**Pull request by h-mayorquin**
In #856 I forgot to scale the timestamps of the spikes correctly. This small PR fixes that by scaling the timestamps of neo (which are usually in seconds) to frames.

I also did some small fixes to some of the docstrings in `neobaseextractor`.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
Ok, so this turned out to be more complicated than I thought. 

The problem is that in the test that was failing:
https://github.com/SpikeInterface/spikeinterface/runs/7651689921?check_suite_focus=true

The corresponding blackrock file has more than one stream. When that happens, the function from neo that extract `_t_start` requires the stream_index to be passed as well but that was not available hence the error in CI:
https://github.com/SpikeInterface/spikeinterface/blob/4346eff5706408435e8aa8345be22fdd452afa27/spikeinterface/extractors/neoextractors/neobaseextractor.py#L172

Therefore, this PR also adds machinery so that when the function `_auto_guess_sampling_frequency(self)` extracts the frequency from the corresponding signal the `signal_id` (from which we can calculate the `signal_index`) is also extracted.

File: spikeinterface/extractors/neoextractors/neobaseextractor.py [hunk 1]
```diff
@@ -229,12 +236,12 @@ def get_unit_spike_train(self, unit_id, start_frame, end_frame):
                                                                 seg_index=self.segment_index,
                                                                 spike_channel_index=unit_index)
         if self._t_start is None:
-            # because handle_spike_frame_directly=True
+            # When handle_spike_frame_directly=True, the extractors handles timestamps as frames
             spike_frames = spike_timestamps
         else:
-            # convert to second second
+            # Conver timestamps to seconds
```
Comment (alejoe91) [inline spikeinterface/extractors/neoextractors/neobaseextractor.py:242]:
```suggestion
            # Convert timestamps to seconds
```

**Comment by h-mayorquin** [comment]
@alejoe91 anything missing here?

**Review by samuelgarcia — APPROVED** [review]

**Comment by samuelgarcia** [comment]
Hi Ramon,
This is OK for me.
Could you rebase to the actual master ?
Be carrfull Alessio some refactoring on neobase extractor.

**Pull request merged**